src/adapter/src/coord/peek.rs

```diff
@@ -202,12 +203,46 @@ pub fn create_fast_path_plan<T: timely::progress::Timestamp>(
                 mir.typ(),
             )));
         } else {
+            if let MirRelationExpr::Project { input, outputs } = mir {
```

Comment:
Should this be a loop to jump through layers of these? Why does this need to be done here instead of in an optimizer step?

Reply:
Hmm, yes! Actually, I just realized that it's being removed by `CanonicalizeMfp` already. I'll delete this code.

Reply:
Done